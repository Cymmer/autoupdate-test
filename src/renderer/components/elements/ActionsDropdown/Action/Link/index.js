import React from 'react';
import { Link } from 'react-router-dom';
import cn from 'classnames';
import PropTypes from 'prop-types';
import styles from '../styles.module.scss';

import Icon from '../../../Icon';
import Tooltip from '../../../Tooltip';
import tooltipPlacement from '../../../Tooltip/constants/tooltipPlacement';

const ActionLink = ({
  action: { id, action, icon, text, tooltip, disabled },
}) =>
  tooltip ? (
    <Tooltip content={tooltip} placement={tooltipPlacement.LEFT}>
      <Link
        className={cn(styles.Action, {
          [styles.Action___disabled]: disabled,
          [styles.Action___delete]: icon === 'delete' || icon === 'close',
        })}
        id={id}
        to={action}
      >
        <Icon className={styles.Action_icon} icon={icon} />
        {text}
      </Link>
    </Tooltip>
  ) : (
    <Link
      className={cn(styles.Action, {
        [styles.Action___disabled]: disabled,
        [styles.Action___delete]: icon === 'delete' || icon === 'close',
      })}
      id={id}
      to={action}
    >
      <Icon className={styles.Action_icon} icon={icon} />
      {text}
    </Link>
  );

ActionLink.propTypes = {
  action: PropTypes.shape({
    action: PropTypes.string.isRequired,
    icon: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired,
    disabled: PropTypes.bool,
    tooltip: PropTypes.string,
    id: PropTypes.string,
  }),
};

export default ActionLink;
