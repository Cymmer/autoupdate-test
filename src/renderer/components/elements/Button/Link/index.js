import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import cn from 'classnames';
import GLOBALS from 'codechum-app-globals';

import { Tooltip } from 'components/elements';
import { STUDENT_ROUTES } from 'screen-wrappers/Student/constants';

import styles from '../styles.module.scss';

import Icon from '../../Icon';
import buttonTypes from '../constants/buttonTypes';

const ButtonLink = ({
  children,
  type,
  className,
  to,
  icon,
  id,
  tabIndex,
  disabled,
  onClick,
  isLocked,
}) => {
  const allChildren = (
    <>
      {icon && (
        <Icon
          className={styles.Button___withIcon_icon}
          data-test="icon"
          icon={icon}
        />
      )}
      {children}
    </>
  );

  const link = to.startsWith('http') ? (
    <a
      className={cn(
        className,
        styles.Button___link,
        styles[`Button___${type}`],
        {
          [styles.Button___withIcon]: icon !== null,
          [styles.Button___disabled]: isLocked || disabled,
        }
      )}
      data-test="nativeAnchor"
      href={to}
      id={id}
      rel="noreferrer"
      tabIndex={tabIndex}
      target="_blank"
      onClick={onClick}
    >
      {allChildren}
    </a>
  ) : (
    <Link
      className={cn(
        className,
        styles.Button___link,
        styles[`Button___${type}`],
        {
          [styles.Button___withIcon]: icon !== null,
          [styles.Button___disabled]: isLocked || disabled,
        }
      )}
      data-test="reactRouterLink"
      id={id}
      tabIndex={tabIndex}
      to={to}
      onClick={onClick}
    >
      {allChildren}
    </Link>
  );

  return !isLocked ? (
    link
  ) : (
    <span className={styles.Button_wrapper}>
      {link}
      <Tooltip content={GLOBALS.MESSAGES.PREMIUM_FEATURE_LOCK_MESSAGE}>
        <a href={STUDENT_ROUTES.SHOP} rel="noreferrer" target="_blank">
          <Icon className={styles.Button_lockedIcon} icon="lock" />
        </a>
      </Tooltip>
    </span>
  );
};

ButtonLink.defaultProps = {
  type: buttonTypes.PRIMARY.BLUE,
  className: null,
  id: null,
  icon: null,
  tabIndex: 0,
  disabled: false,
  onClick: null,
  isLocked: false,
};

ButtonLink.propTypes = {
  type: PropTypes.oneOf([
    buttonTypes.PRIMARY.BLUE,
    buttonTypes.PRIMARY.RED,
    buttonTypes.PRIMARY.GREEN,
    buttonTypes.SECONDARY.BLUE,
    buttonTypes.SECONDARY.GREEN,
    buttonTypes.SECONDARY.RED,
    buttonTypes.TEXT.BLUE,
    buttonTypes.TEXT.RED,
    buttonTypes.TEXT.GREEN,
    buttonTypes.TERTIARY,
  ]),
  children: PropTypes.any.isRequired,
  className: PropTypes.string,
  to: PropTypes.string.isRequired,
  id: PropTypes.string,
  icon: PropTypes.string,
  tabIndex: PropTypes.number,
  disabled: PropTypes.bool,
  onClick: PropTypes.func,
  isLocked: PropTypes.bool,
};

export default ButtonLink;
