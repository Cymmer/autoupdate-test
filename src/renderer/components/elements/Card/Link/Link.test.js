import React from 'react';
import { shallow } from 'enzyme';
import Link from './index';

describe('RENDERING - Card > Link', () => {
  it('should render with default props', () => {
    const component = shallow(<Link>Test</Link>);
    expect(component).toMatchSnapshot();
  });
});
