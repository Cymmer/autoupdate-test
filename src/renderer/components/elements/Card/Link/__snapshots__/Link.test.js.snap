// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RENDERING - Card > Link should render with default props 1`] = `
<Link
  className="Card Card___clickable"
  data-test="cardContainer"
  id={null}
  tabIndex={0}
  target={null}
  to="#"
>
  Test
</Link>
`;
